homeassistant/components/deconz/sensor.py

```diff
@@ -243,53 +259,56 @@ class DeconzTemperature(DeconzDevice, SensorEntity):
     """
 
     TYPE = DOMAIN
+    _device: PydeconzSensor
 
-    def __init__(self, device, gateway):
+    def __init__(self, device: PydeconzSensor, gateway: DeconzGateway) -> None:
         """Initialize deCONZ temperature sensor."""
         super().__init__(device, gateway)
 
         self.entity_description = ENTITY_DESCRIPTIONS[Temperature]
         self._attr_name = f"{self._device.name} Temperature"
 
     @property
-    def unique_id(self):
+    def unique_id(self) -> str:
         """Return a unique identifier for this device."""
         return f"{self.serial}-temperature"
 
     @callback
-    def async_update_callback(self):
+    def async_update_callback(self) -> None:
         """Update the sensor's state."""
         keys = {"temperature", "reachable"}
         if self._device.changed_keys.intersection(keys):
             super().async_update_callback()
 
     @property
-    def native_value(self):
+    def native_value(self) -> float:
         """Return the state of the sensor."""
+        assert isinstance(self._device.secondary_temperature, float)
```

Comment:
We should not have to assert the instance type here

Reply:
Fixed